apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/AbstractR4ResourceProvider.java

```diff
@@ -210,7 +239,37 @@ public T read(@IdParam IdType claimId, RequestDetails requestDetails) {
       throw new ResourceNotFoundException(claimId);
     }
 
-    return claimIdType.getTransformer().transform(metricRegistry, claimEntity, includeTaxNumbers);
+    return transformEntity(claimIdType, claimEntity, includeTaxNumbers);
+  }
+
+  /**
+   * Transforms the pac claim entity to the specified type.
+   *
+   * @param claimIdType the claim type
+   * @param claimEntity the claim entity to transform
+   * @param includeTaxNumbers whether to include tax numbers
+   * @return the transformed claim
+   */
+  private T transformEntity(
+      ResourceTypeV2<T, ?> claimIdType, Object claimEntity, boolean includeTaxNumbers) {
+
+    DomainResource response;
+
+    if (claimIdType instanceof ClaimTypeV2<?> && claimIdType.getTypeLabel().equals("fiss")) {
+      response = fissClaimTransformerV2.transform(claimEntity, includeTaxNumbers);
+    } else if (claimIdType instanceof ClaimTypeV2<?> && claimIdType.getTypeLabel().equals("mcs")) {
+      response = mcsClaimTransformerV2.transform(claimEntity, includeTaxNumbers);
+    } else if (claimIdType instanceof ClaimResponseTypeV2<?>
+        && claimIdType.getTypeLabel().equals("fiss")) {
+      response = fissClaimResponseTransformerV2.transform(claimEntity, includeTaxNumbers);
+    } else if (claimIdType instanceof ClaimResponseTypeV2<?>
+        && claimIdType.getTypeLabel().equals("mcs")) {
+      response = mcsClaimResponseTransformerV2.transform(claimEntity, includeTaxNumbers);
+    } else {
+      throw new InvalidRequestException("Invalid claim id type, cannot get claim data");
+    }
+
+    return (T) response;
```

Comment:
This is adding extra complexity.  What is the payoff?  Is it just removing the metric registry argument?

Reply:
No, this is due to removing the transformer that was attached to the enum for ClaimTypeV2, since it was using a static reference to transform. (Similarly removed the transformer from the non-pac enums, where I assume the pattern was cribbed from)

Reply:
Ultimately the question is why are these specific changes necessary?  What was the problem with having the transformer attached to the enum?